scripts/generate_mc_answer_keys.py

```diff
@@ -0,0 +1,590 @@
+    "ota003": {"device": "M7", "from_type": "NMOS", "to_type": "PMOS", "correct_type": "NMOS"},
+    "ota004": {"device": "M4", "from_type": "PMOS", "to_type": "NMOS", "correct_type": "PMOS"},
+    "ota005": {"device": "M2", "from_type": "NMOS", "to_type": "PMOS", "correct_type": "NMOS"},
+    "ota006": {"device": "M6", "from_type": "PMOS", "to_type": "NMOS", "correct_type": "PMOS"},
+    "ota007": {"device": "M3", "from_type": "NMOS", "to_type": "PMOS", "correct_type": "NMOS"},
+    "ota008": {"device": "M1", "from_type": "PMOS", "to_type": "NMOS", "correct_type": "PMOS"},
+    "ota009": {"device": "M5", "from_type": "NMOS", "to_type": "PMOS", "correct_type": "NMOS"},
+    "ota010": {"device": "M4", "from_type": "PMOS", "to_type": "NMOS", "correct_type": "PMOS"},
+    "ota011": {"device": "M7", "from_type": "NMOS", "to_type": "PMOS", "correct_type": "NMOS"},
+    "ota012": {"device": "M3", "from_type": "PMOS", "to_type": "NMOS", "correct_type": "PMOS"},
+}
+
+DEBUGGING_FEEDBACK_POLARITY_ANSWERS = {
+    "feedback001": {"issue": "positive feedback", "fix": "swap opamp inputs"},
+    "feedback002": {"issue": "positive feedback", "fix": "swap opamp inputs"},
+    "feedback003": {"issue": "positive feedback", "fix": "move feedback to inverting terminal"},
+    "feedback004": {"issue": "positive feedback", "fix": "swap opamp inputs"},
+}
+
+ANALYSIS_OTA_DC_GAIN = {
+    "ota001": "gm·ro/2",       # 5T OTA with current mirror load
+    "ota002": "(gm·ro)²/2",    # Telescopic cascode (fully differential)
+    "ota003": "gm·ro/2",       # High-swing current mirror OTA
+    "ota004": "(gm·ro)²/4",    # Two-stage Miller: (gm1·ro1/2) × (gm2·ro2/2) = (gm·ro)²/4
+    "ota005": "(gm·ro)²/2",    # Single-ended telescopic cascode
+    "ota006": "(gm·ro)²/2",    # Single-ended telescopic cascode
+    "ota007": "gm·ro/2",       # Single-stage common source with current mirror load
+    "ota008": "(gm·ro)²/2",    # NMOS cascode with PMOS cascode load
+    "ota009": "(gm·ro)³/2",    # Gain-boosted cascode (still has parallel ro at output)
+    "ota010": "(gm·ro)²/2",    # Folded cascode (fully differential)
+    "ota011": "(gm·ro)²/2",    # Folded cascode (single-ended)
+    "ota012": "gm·ro/2",       # Folded cascode with current mirror load
+}
+
+ANALYSIS_OTA_GBW = {
+    "ota001": "gm/CL",
+    "ota002": "gm/CL",
+    "ota003": "gm1/Cc",
+    "ota004": "gm/CL",
+    "ota005": "gm/CL",
+    "ota006": "gm/CL",
+    "ota007": "gm/CL",
+    "ota008": "gm/CL",
+    "ota009": "gm/CL",
+    "ota010": "gm/CL",
+    "ota011": "gm/CL",
+    "ota012": "gm/CL",
+}
+
+ANALYSIS_FEEDBACK_LOOP_GAIN = {
+    "feedback001": "T = A0",
+    "feedback002": "T = A0",
+    "feedback003": "T = A0",
+    "feedback004": "T = A0",
+}
+
+ANALYSIS_FEEDBACK_BETA = {
+    "feedback001": "β = 1",
+    "feedback002": "β = 1",
+    "feedback003": "β = R1/(R1+R2)",
+    "feedback004": "β = R1/(R1+R2)",
+}
```

Comment:
These are not very interesting MCQ, right? They are all exactly the same. Should probably have some specificity to the topology. For example, require specificity to which transistor's `gm` matters. Like `gm_{1,2} / CL` etc

Reply:
Should shuffle how the devices are named then to ensure understanding